File: 1.0/src/cz/mg/vulkan/PFNvkAcquireNextImageKHR.java
package cz.mg.vulkan;

public class PFNvkAcquireNextImageKHR extends VkFunctionPointer {
    public PFNvkAcquireNextImageKHR() {
    }

    protected PFNvkAcquireNextImageKHR(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkAcquireNextImageKHR(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkAcquireNextImageKHR(long value) {
        setValue(value);
    }

    public PFNvkAcquireNextImageKHR(VkInstance instance) {
        super(instance, new VkString("vkAcquireNextImageKHR"));
    }

    public int call(VkDevice device, VkSwapchainKHR swapchain, long timeout, VkSemaphore semaphore, VkFence fence, VkUInt32 pImageIndex){
        return callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), swapchain != null ? swapchain.getVkAddress() : VkPointer.getNullAddressNative(), timeout, semaphore != null ? semaphore.getVkAddress() : VkPointer.getNullAddressNative(), fence != null ? fence.getVkAddress() : VkPointer.getNullAddressNative(), pImageIndex != null ? pImageIndex.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callNative(long vkaddress, long device, long swapchain, long timeout, long semaphore, long fence, long pImageIndex);

}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkAcquireNextImageKHR.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkAcquireNextImageKHR_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong swapchain, jlong timeout, jlong semaphore, jlong fence, jlong pImageIndex){
    (void)env;
    (void)jc;
    PFN_vkAcquireNextImageKHR f = (PFN_vkAcquireNextImageKHR)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkSwapchainKHR*)jniLongToPointer(swapchain)),
        (uint64_t)timeout,
        *((VkSemaphore*)jniLongToPointer(semaphore)),
        *((VkFence*)jniLongToPointer(fence)),
        ((uint32_t*)jniLongToPointer(pImageIndex))
    );
}

